### PR TITLE
[main][feat] add path v3 as global config

### PR DESCRIPTION
**Pull request by m100alpaca**
## Description
- make path v3 as global

## Why?
- other contract can use this config
- not restrict to liquidation only

### CODE REVIEW

**Review by wow-alpaca — APPROVED** [review]

**Comment by m100alpaca** [comment]
Note:
Since the liquidation already deployed with set path config inside. Currently we develop a separate path config called "PathPCSV3Reader.sol". Therefore, I have to revise and split contract and test files following this:
**Contracts**
- PancakeswapV3IbTokenLiquidationStrategy (Deployed)
- PancakeswapV3IbTokenLiquidationStrategy_WithPathReader (implemented path reader)
- PancakeswapV3TokenLiquidationStrategy (implemented path reader)
**Tests**
- PancakeswapV3IbTokenLiquidationStrategy (Deployed)
- - PancakeswapV3IbTokenLiquidationStrategy_ExecuteLiquidation
- - PancakeswapV3IbTokenLiquidationStrategy_SetConfigs
- PancakeswapV3IbTokenLiquidationStrategy_WithPathReader
- - WithReader_PancakeswapV3IbTokenLiquidationStrategy_ExecuteLiquidation
- - WithReader_PancakeswapV3IbTokenLiquidationStrategy_SetConfigs
- PancakeswapV3TokenLiquidationStrategy
- - PancakeswapV3TokenLiquidationStrategy_ExecuteLiquidation
- - PancakeswapV3TokenLiquidationStrategy_SetConfigs

File: solidity/contracts/reader/interfaces/IPathPCSV3Reader.sol [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: BUSL
+pragma solidity 0.8.19;
+
+interface IPathPCSV3Reader {
```
Comment (alpanaca) [inline solidity/contracts/reader/interfaces/IPathPCSV3Reader.sol:4]:
```suggestion
interface IUniSwapV3PathReader {
```

File: solidity/contracts/reader/PathPCSV3Reader.sol [hunk 2]
```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: BUSL
+pragma solidity 0.8.19;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { LibPath } from "./libraries/LibPath.sol";
+import { IPancakeV3Pool } from "../money-market/interfaces/IPancakeV3Pool.sol";
+import { IPathPCSV3Reader } from "./interfaces/IPathPCSV3Reader.sol";
+
+contract PathPCSV3Reader is IPathPCSV3Reader, Ownable {
```
Comment (alpanaca) [inline solidity/contracts/reader/PathPCSV3Reader.sol:9]:
```suggestion
contract PCSV3PathReader is IPathPCSV3Reader, Ownable {
```

File: solidity/contracts/reader/PCSV3PathReader.sol [hunk 3]
```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: BUSL
+pragma solidity 0.8.19;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { LibPath } from "./libraries/LibPath.sol";
```
Comment (69alpacas) [inline solidity/contracts/reader/PCSV3PathReader.sol:5]: reuse libs?

File: solidity/contracts/money-market/PancakeswapV3TokenLiquidationStrategy.sol [hunk 4]
```diff
@@ -27,6 +28,7 @@ contract PancakeswapV3TokenLiquidationStrategy is ILiquidationStrategy, Ownable
   error PancakeswapV3TokenLiquidationStrategy_NoLiquidity(address tokenA, address tokenB, uint24 fee);
 
   IPancakeSwapRouterV3 internal immutable router;
+  IUniSwapV3PathReader public pathReader;
```
Comment (69alpacas) [inline solidity/contracts/money-market/PancakeswapV3TokenLiquidationStrategy.sol:31]: immutable or modifiable?

Comment (m100alpaca) [reply solidity/contracts/money-market/PancakeswapV3TokenLiquidationStrategy.sol:31]: immutable

**Review by 69alpacas — APPROVED** [review]

**Review by wow-alpaca — APPROVED** [review]

**Pull request merged**